examples/s2_field_combination_extern_rf_train/nrw_crop_extern_s2_workflow_to_datasets_with_timeshift.ipynb

```diff
@@ -0,0 +1,2364 @@
+    "            start=start_dt,\n",
+    "            end=end_dt,\n",
+    "        ),\n",
+    "        resolution=ge.SpatialResolution(\n",
+    "            10.0,\n",
+    "            10.0,\n",
+    "        ),\n",
+    "        srs=\"EPSG:32632\",\n",
+    ")\n",
+    "\n",
+    "query_rect"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We also need a map from category to number. This is necessary since the RF produces class names and we want to store the result as a raster dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "code_id_map = {\n",
+    "    'None': 0,\n",
+    "    'AF': 1,\n",
+    "    'DA': 2,\n",
+    "    'EP': 3,\n",
+    "    'EW': 4,\n",
+    "    'GL': 5,\n",
+    "    'GM': 6,\n",
+    "    'GT': 7,\n",
+    "    'HF': 8,\n",
+    "    'HP': 9,\n",
+    "    'OE': 10,\n",
+    "    'PA': 11,\n",
+    "    'SF': 12,\n",
+    "    'SL': 13,\n",
+    "    'ZP': 14,    \n",
+    "}"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we query all the workflows in parallel and zip the results together. The result is a list of tiles, where each tile contains the monthly Sentinel-2 data for one band and month.\n",
+    "For each tile (list) we apply the RF to the data and store the result as a new tile. The result is a list of tiles, where each tile contains the predicted crop class for one pixel."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "45\n"
+     ]
+    }
+   ],
+   "source": [
+    "from asyncstdlib.itertools import zip_longest\n",
+    "import xarray as xr\n",
+    "\n",
+    "query_keys = list(band_workflows_b.keys())\n",
+    "queries = [\n",
+    "    w.raster_stream(query_rect) for w in band_workflows_b.values()\n",
+    "]\n",
+    "\n",
+    "res_arrays = []\n",
+    "\n",
+    "async for tile_stac in zip_longest(*queries):\n",
+    "    print(len(tile_stac))\n",
+    "\n",
+    "    arr_stack = xr.concat(tile_stac, dim=\"band\")\n",
+    "    \n",
+    "    res_array = xr.zeros_like(arr_stack.isel(band=0), dtype=np.uint8)\n",
+    "\n",
+    "    for x in range(box_size):\n",
+    "        for y in range(box_size):\n",
+    "            \n",
+    "            rf_input = arr_stack[:, x, y].values\n",
+    "            np.nan_to_num(rf_input, copy=False, nan=0, posinf=None, neginf=None)\n",
+    "\n",
+    "            pred_class = clf.predict([rf_input])[0]           \n",
+    "            res_array[x,y] = code_id_map[pred_class]\n",
+    "\n",
+    "    res_arrays.append(res_array)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now print the result as a image:"
```

Comment:
Maybe briefly interpret the results here as well.
The errors seem to be at the boundaries.